src/screens/WebsocketResearchScreen/SignedChatScreen.tsx

```diff
@@ -42,6 +54,21 @@ const SignedChatScreen = () => {
     flatlistRef.current?.scrollToEnd();
   };
 
+  const moveChatToAnon = async () => {
+    try {
+      setLoading(true);
+      await moveToAnonymousChannel({
+        oldChannelId: selectedChannel?.id,
+        targetUserId: memberChat.user_id,
+        source: 'userId'
+      });
+    } catch (e) {
+      console.log('error moving chat to signed channel', e);
+    } finally {
+      setLoading(false);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling is done using console.log which is not a good practice for production code. Consider using a logging service or showing an error message to the user. Also, the function `moveChatToAnon` is async but there is no mechanism to handle if this function is still running when the component unmounts, which could potentially lead to memory leaks or unexpected behavior.

```diff
- console.log('error moving chat to signed channel', e);
+ // Use a logging service or show an error message to the user

// To prevent potential memory leaks or unexpected behavior, consider using a flag to check if the component is still mounted before setting state.
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->